packages/@romejs/compiler/lint/rules/jsx-a11y/noNoninteractiveElementToInteractiveRole.ts

```diff
@@ -0,0 +1,46 @@
+import {Path, TransformExitResult} from "@romejs/compiler";
+import {descriptions} from "@romejs/diagnostics";
+import {
+	getJSXAttribute,
+	hasJSXAttribute,
+	isJSXElement,
+} from "@romejs/js-ast-utils";
+import getJSXElementName from "@romejs/js-ast-utils/getJSXElementName";
+import {
+	isElementInteractive,
+	isRoleInteractive,
+	roles,
+} from "@romejs/compiler/lint/rules/ariaHelpers";
+
+export default {
+	name: "jsxA11YNoNoninteractiveElementToInteractiveRole",
+	enter(path: Path): TransformExitResult {
+		const {node} = path;
+
+		if (isJSXElement(node)) {
+			if (hasJSXAttribute(node, "role")) {
+				const name = getJSXElementName(node);
+				const roleAttribute = getJSXAttribute(node, "role");
+				if (
+					roleAttribute &&
+					roleAttribute.value &&
+					roleAttribute.value.type === "JSStringLiteral"
+				) {
+					const role = roles.get(roleAttribute.value.value);
+					if (role) {
+						if (!isElementInteractive(name) && isRoleInteractive(role)) {
+							path.context.addNodeDiagnostic(
+								node,
```

Comment:
```suggestion
								roleAttribute,
```

This will increase the specificity of the node reported in the diagnostic.